File: GFG/Array/largestPossibleNumber.cpp
// import java.io.*;
// import java.lang.*;
// import java.util.*;

// #!/usr/bin/python -O

// #include <stdio.h>
// #include <stdlib.h>
// #include <string.h>
// #include<stdbool.h>
// #include<limits.h>

#include<iostream>
#include<algorithm>
#include<string>
#include<vector>
using namespace std;


/*

# Author			: @RAJ009F
# Topic or Type 	: GFG/ARRAY
# Problem Statement	: largest number possible
# Description		: 
# Complexity		: 
=======================
#sample output
----------------------

=======================
*/

int myCompare(string x, string y)
{
	string xy = x.append(y);
	string yx = y.append(x);
	
	return xy.compare(yx)>0?1:0;
	
}

void printLargestNum(vector<string> vec)
{
	sort(vec.begin(), vec.end(), myCompare);
	
	for(int i=0; i<vec.size(); i++)
		cout<<vec[i];
	
}

int main()
{
	
	vector<string> arr;
	arr.push_back("54");
    arr.push_back("546");
    arr.push_back("548");
    arr.push_back("60");
	printLargestNum(arr);
	return 0;
}

File: GFG/Array/largestPossibleNumber.java
// import java.io.*;
// import java.lang.*;
// import java.util.*;

#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include<stdbool.h>
/*

# Author			: @RAJ009F
# Topic or Type 	: GFG/ARRAY
# Problem Statement	: circular Sub Array  sum
# Description		: 
# Complexity		: 
=======================
#sample output
----------------------

=======================
*/


class largestPossibleNumber
{
	public static void heapiFy(String str[], int i,  int size)
	{
		int i = max;
		int l =  2*i+1;
		int r =  2*i+2;
		String x = null;
		String y = null;
		if(l<size )
		{
			x =  str[max]+str[l];
			y =  str[l]+str[max];
			
			if(x.compareTo(y) >= 1)
			{
				max = l;
			}
				
		}
		
		
		
	}
	
	public static void sort(String str[],  int n)
	{
		
		for (int i=n/2-1; i>=0; i--)
		{
			heapiFy(str, i, n);
			
			
		}
		
		
		
	}
	
	public static void printlargestNum(String str[], int n)
	{
		
		sort(str, n);
		
	}
	
	
	public static void main(String args[])
	{
		String[] str = {"54", "546", "548", "60"};
		
		printlargestNum(str, str.length);
	}
	
}